src/Symfony/Component/Validator/Constraints/LocaleValidator.php

```diff
@@ -38,8 +38,9 @@ public function validate($value, Constraint $constraint)
 
         $value = (string) $value;
         $locales = Intl::getLocaleBundle()->getLocaleNames();
+        $aliases = Intl::getLocaleBundle()->getAliases();
 
-        if (!isset($locales[$value])) {
+        if (!isset($locales[$value]) && !array_search($value, $aliases)) {
```

Comment:
As we are not interested in the index wouldn't it be make sense to use `in_array()` instead?